internal/mode/static/nginx/config/generator_test.go

```diff
@@ -88,12 +88,17 @@ func TestGenerate(t *testing.T) {
 	g.Expect(httpCfg).To(ContainSubstring("upstream"))
 	g.Expect(httpCfg).To(ContainSubstring("split_clients"))
 
+	g.Expect(files[2].Path).To(Equal("/etc/nginx/conf.d/matches.json"))
```

Comment:
because in the generator.go we have a range over map
```
	for filepath, bytes := range fileBytes {
		files = append(files, file.File{
			Path:    filepath,
			Content: bytes,
			Type:    file.TypeRegular,
		})
	}
```
the order of `file`s is not deterministic. So, this unit test can fail because it depends on certain order.
I recommended running this test multiple times to see how it can fail.

To prevent that, let's control the order: I suggest sorting files after `files := generator.Generate(conf)`  using sort.Slice using file name.